filters/glitch/src/GlitchFilter.js

```diff
@@ -0,0 +1,269 @@
+        const last = this._slices - 1;
+        const size = this.displacementMapSize;
+        const min = Math.min(this.minSliceWidth / size, 0.9 / this._slices);
+
+        if (this.average) {
+            const count = this._slices;
+            let rest = 1;
+
+            for (let i = 0; i < last; i++) {
+                const average = rest / (count - i);
+                const v =  Math.max(average * (1 - Math.random() * 0.6), min);
+                arr[i] = v;
+                rest -= v;
+            }
+            arr[last] = rest;
+        }
+        else {
+            let rest = 1;
+            const ratio = Math.sqrt(1 / this._slices);
+
+            for (let i = 0; i < last; i++) {
+                const v = Math.max(ratio * rest * Math.random(), min);
+                arr[i] = v;
+                rest -= v;
+            }
+            arr[last] = rest;
+        }
+
+        this.shuffle();
+    }
+
+    initSlicesOffset() {
+        for (let i = 0 ; i < this._slices; i++) {
+            this.slicesOffset[i] = Math.random() * (Math.random() < 0.5 ? -1 : 1);
+        }
+    }
+
+    shuffle() {
+        const arr = this.slicesWidth;
+
+        for (let i = this._slices - 1; i > 0; i--) {
+            const rand = (Math.random() * i) >> 0;
+            const temp = arr[i];
+
+            arr[i] = arr[rand];
+            arr[rand] = temp;
+        }
+    }
+
+    refresh() {
+        this.slicesWidth = new Float32Array(this._slices);
+        this.initSlicesWidth(this.average);
+        this.uniforms.slicesWidth = this.slicesWidth;
+
+        this.slicesOffset = new Float32Array(this._slices);
+        this.initSlicesOffset();
+        this.uniforms.slicesOffset = this.slicesOffset;
+
+        this.updateDisplacementMap();
+    }
+
+    updateDisplacementMap() {
+        let canvas = this.displacementMapCanvas;
+        let size = this.displacementMapSize;
+
+        let ctx = canvas.getContext('2d');
+        ctx.clearRect(0, 0, 8, size);
+
+        let offset;
+        let y = 0;
+        for (let i = 0 ; i < this._slices; i++) {
+            offset = Math.floor(this.slicesOffset[i] * 256);
+            const height = this.slicesWidth[i] * size;
+            const red = offset > 0 ? offset : 0;
+            const green = offset < 0 ? -offset : 0;
+            ctx.fillStyle = 'rgba(' + red + ', ' + green + ', 0, 1)';
+            ctx.fillRect(0, y >> 0, size, height + 1 >> 0);
+            y += height;
+        }
+
+        this.displacementMap._updateID++;
+        this.displacementMap.baseTexture.emit('update', this.displacementMap.baseTexture);
+        // this.displacementMap.update();
+        this.uniforms.displacementMap = this.displacementMap;
+    }
+
+    setSlicesWidth(slicesWidth) {
+        const len = Math.min(this._slices, slicesWidth.length);
+
+        for (let i = 0; i < len; i++){
+            this.slicesWidth[i] = slicesWidth[i];
+        }
+    }
+
+    setSlicesOffset(slicesOffset) {
```

Comment:
`setSlicesOffset`, `setSlicesWidth`, `updateDisplacementMap`, `refresh`, `shuffle`, and `initSlicesOffset` need documentation. What of these methods are private or public?

Reply:
They are public , and for Advanced Usage.
User could custom the effect of giltch (default effect is random generation).

But I don't know how to explain them....

Reply:
Could you at least try to explain and add the `/** */` JSDoc blocks? I can help cleanup the language.